### ISSUE TITLE
`Kpoints` objects using wrong mesh types

### ISSUE THREAD

**Issue by matthewkuner** [materialsproject/pymatgen#3220]
https://github.com/materialsproject/pymatgen/blob/13e46fdf65ecdb60ddd491dd640f236ed3c5ce7a/pymatgen/io/vasp/inputs.py#L1235

Currently, this code could use Monkhorst-pack schemes for FCC and Face-centered orthorhombic cells. According to the [vasp wiki](https://www.vasp.at/wiki/index.php/KPOINTS#Symmetry_reduction_of_the_mesh), this is not good...

Anyone like @janosh @munrojm know why this is?

**Comment by janosh** [comment]
Tagging @arosen93 as the author.

**Comment by esoteric-ephemera** [comment]
Just to copy over part of my discussion with Matthew here: if a Monkhorst-Pack grid doesn't break the symmetry of the lattice, then everything should be OK. I did check the "gold-standard" MP entry, mp-149 / cubic diamond Si, and the input for the task document has a Monkhorst-pack grid (inappropriately)

VASP will throw an error if the breaking of the lattice point / space group symmetry is severe enough. Custodian should catch this

The easy resolution is if the space group of a structure starts with 'F', force a gamma-centered k-point mesh

**Comment by arosen93** [comment]
What am I the author of? 😅 I don't recall touching this.

**Comment by shyuep** [comment]
@arosen93 Your reputation is such that you are the author of *everything* ! :-)

**Comment by matthewkuner** [comment]
@arosen93 It looks like you did actually write this almost 2 years ago lol: https://github.com/materialsproject/pymatgen/commit/8648d50f746486e194c2355d3c12e6dce8f6dea1

**Comment by arosen93** [comment]
True, I admittedly did add the `automatic_density_by_length` function, which is basically a minor variation of `automatic_density_by_vol` for systems with vacuum in _c_. I didn't want to touch any of the existing logic.

Personally, I don't know much about the Monkhorst-Pack vs. Gamma-centered criteria in VASP (and thought at some point we decided to do Gamma-centered for all MP calculations, but I suppose not). I'll defer to @esoteric-ephemera on the best course of action on this one.

**Comment by matthewkuner** [comment]
Yeah it seems like this code existed in some form at least since 2015: https://github.com/materialsproject/pymatgen/commit/423aa162c70d4791489e92e3ef5559b78bd8f2a7#diff-2b61e131c3f0a2a133effdfb20a2b8b6c84690d55bcd81b6d01ea4ecebdb9b0dL902

so basically since this file was created...

**Comment by janosh** [comment]
@arosen93 My bad, I thought we were still talking about `automatic_density_by_lengths` added in https://github.com/materialsproject/pymatgen/commit/8648d50f746486e194c2355d3c12e6dce8f6dea1. 😅

**Comment by arosen93** [comment]
Happy to take the blame regardless. 😉
